lib/gems/pending/util/extensions/miq-benchmark.rb

```diff
@@ -34,20 +34,26 @@ def self.realtime_block(key, &block)
     end
   end
 
+  def self.thread_unique_identifier
+    # Forks inherit the @@realtime_by_tid and parent/child Thread.current.object_id
+    # are equal, so we need to index into the hash with the pid too.
+    "#{Process.pid}-#{Thread.current.object_id}"
```

Comment:
Note, we can't memoize/cache this because child forks will inherit the cached variable too.  :trollface: